opencog/guile/SchemePrimitive.h

```diff
@@ -144,17 +157,28 @@ class SchemePrimitive : public PrimitiveEnviron
 			B_B,   // return boolean, take boolean
 			B_HI,  // return boolean, take handle and int
 			B_HH,  // return boolean, take handle and handle
+                        B_SDDD,
+                        B_SDII,
+                        B_SHDDD,
+                        B_SIDDD,
 			D_HHT, // return double, take handle, handle, and type
 			D_HHTB,// return double, take handle, handle, and type
+                        D_S,
 			H_H,   // return handle, take handle
 			H_HI,  // return handle, take handle and int
 			H_HH,  // return handle, take handle and handle
 			H_HS,  // return handle, take handle and string
 			H_HHH, // return handle, take handle, handle and Handle
 			H_HTQ, // return handle, take handle, type, and HandleSeq
 			H_HZ,  // return handle, take handle and size_t
+                        H_SDDD,
```

Comment:
I think similar comments are worth adding, to counteract the inner crypticness of that code..

```c++
// return handle, take string and 3 double
```